code/game/machinery/porta_turret/portable_turret.dm

```diff
@@ -53,111 +62,117 @@
 
 	var/attacked = 0		//if set to 1, the turret gets pissed off and shoots at people nearby (unless they have sec access!)
 
-	var/on = 1				//determines if the turret is on
-	var/disabled = 0
+	var/on = TRUE				//determines if the turret is on
 
-	var/shot_sound 			//what sound should play when the turret fires
-	var/eshot_sound			//what sound should play when the emagged turret fires
-
-	var/faction = "neutral"
+	var/faction = "turret" // Same faction mobs will never be shot at, no matter the other settings
 
 	var/datum/effect_system/spark_spread/spark_system	//the spark system, used for generating... sparks?
 
-/obj/machinery/porta_turret/New()
+/obj/machinery/porta_turret/New(loc)
 	..()
 	if(!base)
 		base = src
-	icon_state = "[base_icon_state][off_state]"
+	update_icon()
 	//Sets up a spark system
 	spark_system = new /datum/effect_system/spark_spread
 	spark_system.set_up(5, 0, src)
 	spark_system.attach(src)
 
+	setup()
 	if(has_cover)
 		cover = new /obj/machinery/porta_turret_cover(loc)
 		cover.parent_turret = src
-	setup()
 	if(!has_cover)
 		popUp()
 
-
-/obj/machinery/porta_turret/proc/get_base_icon()
-	if(installation)
-		switch(installation)
+/obj/machinery/porta_turret/update_icon()
+	cut_overlays()
+	if(!anchored)
+		icon_state = "turretCover"
+		return
+	if(stat & BROKEN)
+		icon_state = "[base_icon_state]_broken"
+	else
+		if(powered())
+			if(on && raised)
+				switch(mode)
+					if(TURRET_STUN)
+						icon_state = "[base_icon_state]_stun"
+					if(TURRET_LETHAL)
+						icon_state = "[base_icon_state]_lethal"
+			else
+				icon_state = "[base_icon_state]_off"
+		else
+			icon_state = "[base_icon_state]_unpowered"
+
+
+/obj/machinery/porta_turret/proc/setup(obj/item/weapon/gun/turret_gun)
+	if(stored_gun)
+		qdel(stored_gun)
+		stored_gun = null
+
+	if(installation && !turret_gun)
+		stored_gun = new installation(src)
+	else if (turret_gun)
+		stored_gun = turret_gun
+	
+	var/list/gun_properties = stored_gun.get_turret_properties()
+
+	//required properties
+	stun_projectile = gun_properties["stun_projectile"]
+	stun_projectile_sound = gun_properties["stun_projectile_sound"]
+	lethal_projectile = gun_properties["lethal_projectile"]
+	lethal_projectile_sound = gun_properties["lethal_projectile_sound"]
+	base_icon_state = gun_properties["base_icon_state"]
+	
+	//optional properties
+	if(gun_properties["shot_delay"])
+		shot_delay = gun_properties["shot_delay"]
+	if(gun_properties["reqpower"])
+		reqpower = gun_properties["reqpower"]
+	
+	update_icon()
+	return gun_properties
+		
+		/*

```

Comment:
Oh, whoops. Left this in to see if the sounds fit with oldcode. Will remove in a moment.